### PR TITLE
Add guideline for grouping Marionette view options

### PR DESCRIPTION
**Pull request by traviskroberts**

### CODE REVIEW

File: JavaScript.md [hunk 1]
```diff
@@ -92,18 +92,25 @@ Much of this was taken from [idiomatic-js](https://github.com/necolas/idiomatic-
...
 
   # good
-  Marionette.ItemVew.extend
-    template: tpl
+  Marionette.CollectionVew.extend
     className: 'view-edit-classroom'
+    template: tpl
+
+    itemView: itemView
+    itemViewContainer: '#js-itemview-container'
```
Comment (stevencwarren) [inline JavaScript.md:131]: Do we want to add `ui`, `initialize`, `onRender`, `onShow`, and other stuff?

**Comment by traviskroberts** [comment]
@dcalhoun Can you also sign off on this PR?

**Comment by dcalhoun** [comment]
I didn't follow what "favoring template/DOM attributes above others" means?

Also, it might be worthwhile define what "groups of related properties" are, i.e.:
1. View props (alpha)
2. Child view props (alpha)
3. Marionette methods (lifecycle?)
4. Added methods (alpha)

Not necessary, just a thought. Looks good. :+1:

**Comment by stevencwarren** [comment]
:+1:

**Comment by dcalhoun** [comment]
Smile

**Comment by traviskroberts** [comment]
Most. Arduous. PR. Ever.

**Pull request merged**